src/commands/detectorDiagnostics.ts

```diff
@@ -0,0 +1,130 @@
+
+  vscode.window.withProgress({
+    location: vscode.ProgressLocation.Notification,
+    title: `Loading ${clsutername} diagnostics.`,
+    cancellable: false
+  }, () => {
+    return new Promise(async (resolve) => {
+      const clusterAppLensData = await getAppLensDetectorData(cloudTarget);
+
+      if (clusterAppLensData) {
+        await createDetectorWebView(clsutername, clusterAppLensData, extensionPath);
+      }
+      resolve();
+    });
+  });
+}
+
+async function createDetectorWebView(
+  clusterName: string,
+  clusterAppLensData: AppLensARMresponse,
+  extensionPath: string) {
+    const panel = vscode.window.createWebviewPanel(
+      'AKS Diagnostics',
+      'AKS diagnostics view for: ' + clusterName,
+      vscode.ViewColumn.Active,
+      {
+        enableScripts: true,
+        enableCommandUris: true
+      }
+    );
+
+    panel.webview.html = getWebviewContent(clusterAppLensData, extensionPath);
+}
+
+async function getAppLensDetectorData(
+  clusterTarget: k8s.CloudExplorerV1.CloudExplorerResourceNode
+  ): Promise<AppLensARMresponse | undefined> {
+  const apiResult = await getClusterInsightInfo(clusterTarget);
+
+  if (apiResult.succeeded) {
+    return apiResult.result.apiresult;
+  } else if (!apiResult.succeeded) {
+    vscode.window.showInformationMessage(apiResult.error);
+  }
+  return;
+}
+
+async function getClusterInsightInfo(
+  target: k8s.CloudExplorerV1.CloudExplorerResourceNode
+  ): Promise<Errorable<AppLensAPIResult>> {
+  try {
+      const client = new ResourceManagementClient(target.cloudResource.root.credentials, target.cloudResource.root.subscriptionId);
+      const clusterInsightInfo = await client.resources.get(
+        target.cloudResource.armId.split("/")[4], target.cloudResource.resource.type,
+        target.cloudResource.resource.name, "detectors", "mcrEndpointUpdate", "2019-08-01");
+
+      return { succeeded: true, result: {apiresult: <AppLensARMresponse> clusterInsightInfo}};
+    } catch (ex) {
+      return { succeeded: false, error:  `Error invoking cluster insight: ${ex}` };
+    }
+}
+
+function getWebviewContent(
+  clusterdata: AppLensARMresponse,
+  vscodeExtensionPath: string) {
+    const webviewClusterData = clusterdata?.properties;
+
+    const stylePathOnDisk = vscode.Uri.file(path.join(vscodeExtensionPath, 'src', 'commands', 'style', 'detector.css'));
+    const pathToHtmlOnDisk = vscode.Uri.file(path.join(vscodeExtensionPath, 'src', 'commands', 'style', 'detector.html'));
+    const styleUri = stylePathOnDisk.with({ scheme: 'vscode-resource' });
+    const pathUri = pathToHtmlOnDisk.with({scheme: 'vscode-resource'});
+
+    const htmldata = fs.readFileSync(pathUri.fsPath, 'utf8').toString();
+    const template = htmlhandlers.compile(htmldata);
+    const data = { "cssuri": `${styleUri}`,
+                   "clustername": `${webviewClusterData.metadata.name}`,
+                   "rowdata": `${webviewClusterData.dataset[0].table.rows[1].toString().split(',')[3]}`
```

Comment:
Definitely pull this out into a named helper because I have _no idea_ what the significance of the 0s, 1s and 3s is!

Reply:
+1 and this is part which will be clear soon, hopefully once `AppLens` API is fixed.

Reply:
💡 `AppLens API` update just as a note: AppLens folks are still sixing their end of API and data contract et. al. (hopefully stays same. 📓 Which means once very little from vscod eside of things are left and once we have clear API we can move forward with the end-goal. Thanks.

Reply:
💡 **keeping this up-to-date**:  `AppLens` API is not consumable for this extension purpose hence we are figuring out what detector api to expose for this experience.

Reply:
💡 API discussion is over: we will be using `detectors` - `networkconnectivity` arm api.